src/redisraft.c

```diff
@@ -1874,6 +1883,18 @@ static int registerRaftCommands(RedisModuleCtx *ctx)
     return REDISMODULE_OK;
 }
 
+void moduleChangeCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub, void *data)
+{
+    REDISMODULE_NOT_USED(e);
+
+    if (sub != REDISMODULE_SUBEVENT_MODULE_LOADED) {
```

Comment:
We probably don't need to remove cmds from our dict, but we do want to update them in case a new version of a module was loaded - in the current implementation the dict will not be updated with the new flags.

Reply:
We should call this also on `REDISMODULE_SUBEVENT_MODULE_UNLOADED`. 
So, it will refresh the command table. On load, it will add new specs. On unload, it will remove stale specs. Wdyt?

Reply:
yea, the condition is wrong, unsure what I was thinking.  need to fix.

Reply:
Agree, I guess there is no really need to do any optimization here :)